app/controllers/stories_controller.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+class StoriesController < ApplicationController
+  before_action :update_top_stories, only: :index
+
+  def index
+    @stories = Story.where(hacker_news_id: top_stories_ids).includes(:users)
```

Comment:
The `users` are the users who have starred this article, include them to avoid an n+1.